plugins/inputs/docker/docker.go

```diff
@@ -565,18 +599,60 @@ func (d *Docker) gatherContainerInspect(
 		}
 	}
 
-	parseContainerStats(v, acc, tags, container.ID, d.PerDevice, d.Total, daemonOSType)
+	// Temporary logic needed for backwards compatibility until 'perdevice' setting is removed.
+	perDeviceInclude := getEffectivePerDeviceInclude(d.PerDevice, d.PerDeviceInclude, d.Log)
+	// Temporary logic needed for backwards compatibility until 'total' setting is removed.
+	totalInclude := getEffectiveTotalInclude(d.Total, d.TotalInclude, d.Log)
+
+	parseContainerStats(v, acc, tags, container.ID, perDeviceInclude, totalInclude, daemonOSType)
 
 	return nil
 }
 
+// getEffectivePerdeviceInclude returns the effective 'perdevice_include' setting to be applied.
+// This is a temporary logic to keep backwards compatibility and should be removed once 'perdevice' is deprecated.
+// If 'perdevice' is 'true', 'network' and 'blkio' are unconditionally added.
+func getEffectivePerDeviceInclude(perDevice bool, perDeviceInclude []string,
+	log telegraf.Logger) (effectivePerDeviceInclude []string) {
+	if !perDevice {
+		effectivePerDeviceInclude = perDeviceInclude
+	} else {
+		log.Warn("'perdevice' setting is set to 'true' so 'blkio' and 'network' metrics will be collected. " +
+			"Please set it to 'false' and use 'perdevice_include' instead to control this behaviour as 'perdevice' " +
+			"will be deprecated")
+		if choice.Contains("cpu", perDeviceInclude) {
+			effectivePerDeviceInclude = append(effectivePerDeviceInclude, "cpu")
+		}
+
+		effectivePerDeviceInclude = append(effectivePerDeviceInclude, "network", "blkio")
+	}
+	return
+}
```

Comment:
You might ease your life drastically if you make this a `(d *Docker)` member function. This way you could write:
```suggestion
func (d *Docker) getEffectivePerDeviceInclude() {
	if perDevice {
		d.Log.Warn("'perdevice' setting is set to 'true' so 'blkio' and 'network' metrics will be collected. " +
			"Please set it to 'false' and use 'perdevice_include' instead to control this behaviour as 'perdevice' " +
			"will be deprecated")
		d.PerDeviceInclude = []string{"network", "blkio")  // Or whatever the current default is
	}
}
```

These few line could also directly be added to the `Init()` function... Furthermore, you could define the deprecation warning string as constant somewhere at the top of the file so we can shorten the code.

Reply:
Yup, resultant code is definitely shorter, thx!